### ISSUE TITLE
Packaging nix-installer for openSUSE: How to set or where to get the DETERMINATE_NIXD_BINARY_PATH?

### ISSUE THREAD

**Issue by johanneskastl** [DeterminateSystems/nix-installer#1268]
Dear maintainers,

I packaged the nix-installer for openSUSE and got a working version for 0.20.1. But starting with 0.21.0, it errors out due to `DETERMINATE_NIXD_BINARY_PATH` being unset.

I had a look around, but none of the values I found seem to work (see error below). I am guessing I need to download the nixd binary upfront, just like I need to do with the tarballs for `NIX_INSTALLER_TARBALL_PATH`.

Setting this to an empty value did not work. So I guess I have to set a proper value. But which?

Question: Where can I obtain the proper nixd binary? To what values do I need to set this?

Thanks in advance!

Kind Regards,
Johannes

P.S.: Example output when setting `DETERMINATE_NIXD_BINARY_PATH=/nix/determinate/determinate-nixd`:

```
[  212s]    Compiling nix-installer v0.21.0 (/home/abuild/rpmbuild/BUILD/nix-installer-0.21.0)
[  215s] error: couldn't read `/nix/determinate/determinate-nixd`: No such file or directory (os error 2)
[  215s]   --> src/settings.rs:29:10
[  215s]    |
[  215s] 29 |     Some(include_bytes!(env!("DETERMINATE_NIXD_BINARY_PATH")));
[  215s]    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[  215s]    |
[  215s]    = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)
[  215s] 
[  230s] error: could not compile `nix-installer` (lib) due to 1 previous error
[  230s] error: Bad exit status from /var/tmp/rpm-tmp.s6efYx (%build)
```

**Comment by cole-h** [comment]
We get these binaries via the `determinate` flake input. They need to be the actual binaries, since we embed them into the `nix-installer`. The binary is also different per architecture (except for macOS where it's combined into the same binary). You can find the URLs here: https://github.com/DeterminateSystems/determinate/blob/afd03f6548b1e86c2f2aa23c35a7672221605b82/flake.nix#L8-L20

The inclusion of this path is controlled by the `determinate-nix` feature: https://github.com/DeterminateSystems/nix-installer/blob/7fcb69bb90602199ea397460f18b56d201e914f3/src/settings.rs#L21

**Comment by johanneskastl** [comment]
Thanks @cole-h , that helps.

I guess the `DETERMINATE_NIXD_BINARY_PATH` environment variable should be set to the proper (arch-dependent) file name just like the `NIX_INSTALLER_TARBALL_PATH` one during compile time?

I'll test and report back, if this does not work.

**Comment by cole-h** [comment]
Yep, exactly!

**Comment by johanneskastl** [comment]
One more question: Does it always use the latest commit in the `main` branch of https://github.com/DeterminateSystems/determinate as source for the determinate flake? Or is there another relation between nix-installer versions and the nixd binary?

**Comment by cole-h** [comment]
No. The version it's currently using is pinned in the flake.lock: https://github.com/DeterminateSystems/nix-installer/blob/7fcb69bb90602199ea397460f18b56d201e914f3/flake.lock#L32-L67
